### PR TITLE
Custom Streamfield

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1418582836)

### Description of Changes Made

This PR makes changes to the way StreamField migrations are handled on the project. This is done by introducing a custom StreamField class `tbx.core.utils.fields.StreamField`.

This feature was borrowed from the work done by @ababic (and other contributors) in _kit_. 🫡

#### Wagtail admin tests

- Existing StreamField content shouldn't be broken
  - Content should continue to be rendered as before
  - When you edit existing StreamField content, you should be able to see the changes you made, as expected

#### Dev tests

- If you make changes to any StreamField blocks and run `./manage.py makemigrations --dry-run`, there shouldn't be any migrations

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [x] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on Linux
  - Latest version of Firefox on Linux
- [ ] Not required

#### Data protection

- [ ] Not relevant
- [x] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

**Comment by nimasmi** [comment]
Thanks @engineervix, the bits which modify StreamField are as described in the ticket and the MR description, but the custom migration monkey patching is not mentioned.

Coincidentally this was also discussed today at https://torchbox.slack.com/archives/C03R0FNK7/p1710251769524659?thread_ts=1710247414.188509&cid=C03R0FNK7, and I'm chasing there to verify that these MRs are indeed distinct. I think Wagtail Kit is lacking documentation of the monkey patching changes.

**Comment by engineervix** [comment]
> Thanks @engineervix, the bits which modify StreamField are as described in the ticket and the MR description, but the custom migration monkey patching is not mentioned.
> 
> Coincidentally this was also discussed today at https://torchbox.slack.com/archives/C03R0FNK7/p1710251769524659?thread_ts=1710247414.188509&cid=C03R0FNK7, and I'm chasing there to verify that these MRs are indeed distinct. I think Wagtail Kit is lacking documentation of the monkey patching changes.

Thanks @nimasmi I've updated the MR to remove the migration monkeypatching

File: docs/upgrading.md [hunk 1]
```diff
@@ -26,7 +26,13 @@ The following areas of functionality are critical paths for the site which don't
...
+### General
+
 - Other places where you know extra maintenance or checks may be necessary
 - This could be code which you know should be checked and possibly removed - e.g. because you've patched something until a fix is merged in a subsequent release.
 - Any previous fixes which may need to be updated/reapplied on subsequent upgrades
 - Technical debt which could be affected by an upgrade.
+
+### Custom StreamField
+
+As indicated [here](./custom-features/migration-friendly-streamfields.md), this project uses a custom field class (`tbx.core.utils.fields.StreamField`) instead of the usual `wagtail.fields.StreamField` field for streamfield content.
```
Comment (nimasmi) [inline docs/upgrading.md:38]: **Praise**: Good shout to include it here. Can you backport this to Wagtail Starter Kit, if it's not already there?

**Review by nimasmi — APPROVED** [review]
Thanks @engineervix, I approve.

**Pull request merged**